### PR TITLE
Fix mutations minor typos

### PR DESCRIPTION
**Pull request by Faalagorn**
#### Summary
None

#### Purpose of change
Fix some errors spotted in Transifex.

#### Describe the solution
Fix mutation typos spotted in Transifex.

#### Describe alternatives you've considered
Add actual description for survivor stories (these should only be seen in debug though). 

#### Testing
Compiles fine, changes can be seen in debug menu by adding mutations in question.

### CODE REVIEW

File: data/json/npcs/BG_traits.json [hunk 1]
```diff
@@ -697,9 +697,9 @@
     "type": "mutation",
     "//": "This NPC cannot tell you about other NPCs.",
     "id": "BOSS_No_Others_01",
-    "name": { "str": "Other survivors story" },
+    "name": { "str": "Other Survivors Story" },
```
Comment (haveric) [inline data/json/npcs/BG_traits.json:700]: Should these have an apostrophe somewhere on Survivors?

Comment (Faalagorn) [reply data/json/npcs/BG_traits.json:700]: I was thinking about it too; whether it should be "Other Survivor Story", "Other Survivor's Story" or "Other Survivors Story" (plural).

Comment (haveric) [reply data/json/npcs/BG_traits.json:700]: I don't know the context of these, so I wasn't sure which, if any it should have, but figured it was worth mentioning

Comment (LyleSY) [reply data/json/npcs/BG_traits.json:700]: I was one of the authors on this, these mutations determine whether or not an NPC knows about other survivors. The current formatting is correct, though capitalization is fine

**Review by LyleSY — APPROVED** [review]

**Pull request merged**